crates/ruff_linter/src/rules/airflow/rules/moved_to_provider_in_3.rs

```diff
@@ -0,0 +1,150 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Expr, ExprAttribute};
+use ruff_python_semantic::Modules;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+#[derive(Debug, Eq, PartialEq)]
+enum Replacement {
+    ProviderName(String, String, String),
+    ImportPathMoved(String, String, String, String),
+}
```

Comment:
Do you for see cases where any of the strings are dynamic (contain e.g. an expression)? I'd otherwise suggest to use `&'static str` here to avoid the annoying `to_string` calls and reduce the amount of allocations
```suggestion
enum Replacement {
    ProviderName(&'static str, &'static str &'static str),
    ImportPathMoved(&'static str, &'static str, &'static str, &'static str),
}
```

Reply:
I guess not 🤔  Thanks for your suggestion! I'll change it this way. we can change it back if we really need to do that.

Reply:
fixed. Thanks!